mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -1,200 +1,178 @@
+        }),
+    );
+    commands.insert(
+        AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        Box::new(AuthorizedIdentities {
+            topic: AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        }),
+    );
+    commands
+}
 
-    let mut subacks: HashSet<_> = topics.iter().map(Clone::clone).collect();
-
-    while let Some(event) = client
-        .try_next()
-        .await
-        .map_err(CommandHandlerError::PollClientFailure)?
-    {
-        if let Event::SubscriptionUpdates(subscriptions) = event {
-            for subscription in subscriptions {
-                if let SubscriptionUpdateEvent::Subscribe(sub) = subscription {
-                    subacks.remove(&sub.topic_filter);
-                }
-            }
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ServiceIdentity {
+    #[serde(rename = "Identity")]
+    identity: String,
+    #[serde(rename = "AuthChain")]
+    auth_chain: Option<String>,
+}
 
-            if subacks.is_empty() {
-                debug!("command handler successfully subscribed to disconnect topic");
-                return Ok(());
+impl fmt::Display for ServiceIdentity {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self.auth_chain {
+            Some(auth_chain) => {
+                write!(f, "Identity: {}; Auth_Chain: {}", self.identity, auth_chain)
             }
+            None => write!(f, "Identity: {}", self.identity),
         }
     }
-
-    error!("command handler failed to subscribe to disconnect topic");
-    Err(CommandHandlerError::MissingSubacks(
-        subacks.into_iter().collect::<Vec<_>>(),
-    ))
 }
 
 #[derive(Debug, thiserror::Error)]
-pub enum CommandHandlerError {
-    #[error("failed to receive expected subacks for command topics: {0:?}")]
-    MissingSubacks(Vec<String>),
+pub enum HandleEventError {
+    #[error("failed to parse client id from message payload")]
+    ParseClientId(SerdeError),
 
-    #[error("failed to subscribe command handler to command topic")]
-    SubscribeFailure(#[from] UpdateSubscriptionError),
+    #[error("failed to parse authorized identities from message payload")]
+    ParseAuthorizedIdentities(SerdeError),
 
-    #[error("failed to poll client when validating command handler subscriptions")]
-    PollClientFailure(#[from] mqtt3::Error),
+    #[error("failed while sending authorized identities to broker")]
+    SendAuthorizedIdentitiesToBroker(Error),
 
-    #[error("failed to signal shutdown for command handler")]
-    ShutdownClient(#[from] mqtt3::ShutdownError),
+    #[error("failed while sending broker signal to disconnect client")]
+    DisconnectSignal(Error),
 }
 
-#[derive(Debug, thiserror::Error)]
-enum HandleDisconnectError {
-    #[error("failed to parse client id from message payload")]
-    ParseClientId(#[from] SerdeError),
+#[cfg(test)]
+mod tests {
+    use crate::command::ServiceIdentity;
+
+    #[test]
+    fn deserialize_broker_service_identity() {
```

Comment:
to be honest I'm not sure if these tests have value. You are basically testing serde_json, which is 3rd party crate and already well-tested. What you could do, is to add integration tests (similar to what we have for disconnect command and topic translation). The test case could be:
- Create a broker
- Send an update to a auth command with one or few service identities
- Connect client with this identity (should be successful)
- Send an update where indentity is no longer there
- Expected: Client should be disconnected.

You can add this test at the end, when you implement system event handler in the broker

Reply:
Yes makes sense. At least they were helpful to prove to myself that the serde works as expected :)